gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/transformer/RequestPreProcessor.java

```diff
@@ -109,10 +110,23 @@ public EntitiesRequest transform(
         transformFilter(attributeIdMappings, originalRequest.getFilter(), context);
     Filter filter = mergeFilters(filterToInject, transformedFilter);
 
+    // Convert the time range into a filter and set it on the request so that all downstream
```

Comment:
@tim-mwangi @aaron-steinfeld @avinashkolluru This is the change I think fixes the bug around entity filtering when time range is involved and also keeps the query execution tree code simpler. 
Wanted to get your inputs on this before I go deep down this approach.

Reply:
I don't know what all goes on in the query execution tree as to why this is actually needed, but my concern would be that there are now two sources of truth WRT the time range. Code that needs to directly work with the time range (for example, the bucket alignment code, if that's still needed), would need to either read from the first class field, or add significant complexity to read from the filter tree.

Maybe it would help if you could show where the code gets simpler as a result of this?

Reply:
Updated the PR description with details and also logged https://github.com/hypertrace/hypertrace/issues/119 to fix the tech-debt to clear confusion.